### DISCUSSION TITLE
Salesforce Commerce Cloud Front End Debug Using Visual Studio Code

### DISCUSSION THREAD

**Discussion by Gabenny** [SalesforceCommerceCloud/sfcc-ci#527] [Team Posts]
Hi,

I'm a Salesforce Commerce Cloud developer and I would like to know if you can give me some hints on Front End JS code debugging using Visual Studio Code.

Actually I'm using Prophet debugger Visual Studio Code Plugin to debug Back End code including JS and DS files.

The best scenario for me is to be able to debug Front End and Back End Code at the same time using a single plateform with is Visual Studio Code.

Can you please help me find out a solution?

Thank you.

Best regards

**Comment by pravngaur** [comment]
Prophet debugger plugin is for Node js debugging when tied with SFCC
environments... For FE js debug you would need to use other plugins like
https://raygun.com/blog/debug-javascript-visual-studio/ (i have not used
this plugin personally)

But this would still be different plugins, that i think would be harder to
manage than prophet and chrome debug tool...

**Comment by nareshnegikumar** [comment]
For FE JS debug, better to stick with Chrome debugger tool.

Still want everything in VS install “Debugger for Chrome” extension to get all feature in VS.


Thanks
Naresh Negi

**Comment by Gabenny** [comment]
Hi,

Thank you for your reply.

I think that the best tool to use could be chrome debug tool.

But still I'm novice on this plugging and I'm still trying to figure out how to connect my  SFCCsandbox instance to chrome debug tool.

Can you please share with if possible config files with attachement to SFCC Sandbox?

Thank you

Best regards

**Comment by nimitshukla123** [comment]
I think for using chrome debug tool you application should run locally.

Thanks
Nimit Shukla

On Mon, Mar 25, 2019 at 8:24 PM Amine Gabriel Bennani <
notifications@github.com> wrote:

> Hi,
>
> Thank you for your reply.
>
> I think that the best tool to use could be chrome debug tool.
>
> But still I'm novice on this plugging and I'm still trying to figure out
> how to connect my SFCCsandbox instance to chrome debug tool.
>
> Can you please share with if possible config files with attachement to
> SFCC Sandbox?
>
> Thank you
>
> Best regards
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/14/comments/3>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANi2X7JgA5CtKY_tkybBPtzctdB0wAtuks5vaL_wgaJpZM4cGr2C>
> .
>


-- 
Thanks & Regards,

Nimit Shukla
Mobile: +91-9599943809

**Comment by pravngaur** [comment]
Hello Amine,

You need to deploy storefront to SFCC sandbox in order to deploy your
application..

You can do it using either:

1) Npm uploadCartridge command - if you are using SFRA.
2) or you can connect your IDE to upload the cartridges, in VS code - you
need to cobfigure Prophet debugger, while in eclise its just the matter of
successfully connecting to sandbox.

Once the application is deployed, open the storefront in chrome browser and
start using the debug tools, for js debugging check the console tab...

**Comment by ericjohnson-sfdc** [comment]
This tutorial seems like a good way to learn basic front-end debugging with VS Code and the Chrome debugger plugin.

https://www.sitepoint.com/debugging-javascript-projects-vs-code-chrome-debugger/
